pkg/ingester/ingester_ring.go

```diff
@@ -119,7 +125,7 @@ func (cfg *RingConfig) ToRingConfig() ring.Config {
 
 // ToLifecyclerConfig returns a ring.LifecyclerConfig based on the ingester
 // ring config.
-func (cfg *RingConfig) ToLifecyclerConfig() ring.LifecyclerConfig {
+func (cfg *RingConfig) ToLifecyclerConfig(logger log.Logger) ring.LifecyclerConfig {
```

Comment:
I have put the check in this method since it is where the flag is consumed, Another way is to implement a validate method of RingConfig, and call it from mimir.go